src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -73,24 +157,132 @@ export const TestEventDetailsCache = createCache<
+
+export const testEventDetailsResultsCache: ExternallyManagedCache<
+  [executionPoint: ExecutionPoint],
+  TestEventDetailsEntry
+> = createExternallyManagedCache({
+  debugLabel: `TestEventDetailsResultsCache`,
+  getKey: ([executionPoint, ...params]) => {
+    return executionPoint;
   },
 });
+
+export const testEventDomNodeCache: ExternallyManagedCache<
+  [executionPoint: ExecutionPoint],
+  TestEventDomNodeDetails
+> = createExternallyManagedCache({
+  debugLabel: `TestEventDetailsResultsCache`,
+  getKey: ([executionPoint, ...params]) => {
+    return executionPoint;
+  },
+});
+
+// Pre-fetch the DOM node for this step, if possible.
+// Note this runs somewhat in the background and is not awaited,
+// and mutates the `testEventDomNodeCache` with the results.
+async function fetchAndCachePossibleDomNode(
+  replayClient: ReplayClientInterface,
+  sanitized: ProtocolObject,
+  pauseId: string,
+  timeStampedPoint: TimeStampedPoint
+) {
+  const propNamesWithPotentialElements = ["Yielded", "Applied To"] as const;
+
+  const propsWithPotentialElements =
+    sanitized.preview?.properties?.filter(({ name }) =>
+      propNamesWithPotentialElements.includes(name)
+    ) ?? [];
+
+  const possibleDomNodes = await Promise.all(
+    propsWithPotentialElements.map(async prop => {
+      if (!prop.object) {
+        return null;
+      }
+
+      const cachedPropObject = await objectCache.readAsync(
+        replayClient,
+        pauseId,
+        prop.object,
+        "canOverflow"
+      );
+
+      if (cachedPropObject.className === "Array") {
+        // Probably multiple DOM nodes. Pre-fetch these too.
+        const yieldedPropsWithObjects = (cachedPropObject.preview?.properties ?? []).filter(
+          prop => prop.object
+        );
+
+        const yieldedDomNodes = await Promise.all(
+          yieldedPropsWithObjects.map(async (prop: Property) => {
+            const cachedPropObject = await objectCache.readAsync(
+              replayClient,
+              pauseId,
+              prop.object!,
+              "canOverflow"
+            );
+            return cachedPropObject;
+          })
+        );
+        return yieldedDomNodes[0] ?? null;
+      } else {
+        return cachedPropObject;
+      }
+    })
+  );
+
+  // Try to make sure this looks like a DOM node
+  const firstDomNode = possibleDomNodes.find(
+    el => !!el && el.className !== "Object" && el.preview?.node
+  );
+
+  let nodeInfo: NodeInfo | null = null;
+
+  if (firstDomNode) {
+    // Kick off a fetch for the box model now, so we have that cached when we try to highlight this node
+    boxModelCache.prefetch(replayClient, pauseId, firstDomNode.objectId);
+    nodeInfo = await processedNodeDataCache.readAsync(replayClient, pauseId, firstDomNode.objectId);
+  }
+
+  const domNodeDetails: TestEventDomNodeDetails = {
+    ...timeStampedPoint,
+    pauseId,
+    domNode: nodeInfo,
+  };
+
+  testEventDomNodeCache.cacheValue(domNodeDetails, domNodeDetails.point);
```

Comment:
My main reason was to simplify the lookup over in `TestSectionRow`.  It doesn't currently get `testRecording` as a prop, and we'd need that as part of the interval cache params.  Also, this logic was originally inlined as part of the overall interval cache fetching.  I think you're right that now that it's a separate function, this could probably be its own cache.

Reply:
Hmm. I just wrote the code to switch to a normal `createCache`... then realized I have the same problem. The logic needs four args ( `replayClient, sanitizedObject, pauseId, timeStampedPoint`), and we really only _ought_ to have to provide the execution point to read it.  

So I'm going to stick with the external cache for now.